nifi-extension-bundles/nifi-standard-bundle/nifi-standard-rules/src/main/java/org/apache/nifi/flowanalysis/rules/RestrictBackpressureSettings.java

```diff
@@ -0,0 +1,241 @@
+                                idComponent.get(connection.getDestination().getId()),
+                                BackpressureViolationType.BP_SIZE_THRESHOLD_ABOVE_LIMIT,
+                                getViolationMessage(BackpressureViolationType.BP_SIZE_THRESHOLD_ABOVE_LIMIT, connection.getBackPressureDataSizeThreshold(), context.getProperty(SIZE_MAX).getValue())));
+                    }
+                }
+            );
+
+        return results;
+    }
+
+    private GroupAnalysisResult buildViolation(final VersionedConnection connection, final VersionedComponent source,
+            final VersionedComponent destination, final BackpressureViolationType backpressureViolationType, final String violationMessage) {
+        // The reason why we want the violation to be on the processor when we have one
+        // (either as source or destination) is because in case the rule is "enforced"
+        // we want the corresponding component to be invalid. If this is not a processor
+        // (funnel, process group, etc) we cannot make it invalid so we put the
+        // violation on the connection itself.
+        if (!(source instanceof VersionedProcessor) && !(destination instanceof VersionedProcessor)) {
+            // connection between two components that are not processors and cannot be invalid, setting violation on connection
+            return GroupAnalysisResult.forComponent(connection,
+                    connection.getIdentifier() + "_" + backpressureViolationType.getId(),
+                    getLocationMessage(connection, source, destination) + violationMessage).build();
+        } else if (source instanceof VersionedProcessor) {
+            // defining violation on source processor
+            return GroupAnalysisResult.forComponent(source,
+                    connection.getIdentifier() + "_" + backpressureViolationType.getId(),
+                    getLocationMessage(connection, source, destination) + violationMessage).build();
+        } else {
+            // defining violation on destination processor
+            return GroupAnalysisResult.forComponent(destination,
+                    connection.getIdentifier() + "_" + backpressureViolationType.getId(),
+                    getLocationMessage(connection, source, destination) + violationMessage).build();
+        }
+    }
+
+    private String getLocationMessage(final VersionedConnection connection, final VersionedComponent source, final VersionedComponent destination) {
+        if (source == null || destination == null) {
+            return "The connection [" + connection.getIdentifier() + "] is violating the rule for backpressure settings. ";
+        }
+        return "The connection [" + connection.getIdentifier() + "] connecting " + source.getName() + " [" + source.getIdentifier() + "] to "
+                + destination.getName() + " [" + destination.getIdentifier() + "] is violating the rule for backpressure settings. ";
+    }
```

Comment:
I feel like this can be improved, because there are cases where one source/destination exists and the other is null.  I'm thinking one of:
- "The connection [] is violating the rule"
- "The connection [] from source [] is violating the rule"
- "The connection [] to destination [] is violating the rule"
- "The connection [] from source [] to destination [] is violating the rule"

I have also found that when the source or destination is a Funnel and the other is a Processor, that the Funnel is referred to as "null", such as "The connection [] connecting UpdateAttribute [] to null [] is violating the rule".  We can probably detect that the VersionedComponent is a funnel (that has no name) and replace the name with the word "Funnel".

Reply:
If you would like, we can accept this PR as it is, and I can improve this message while I'm working on a new FlowAnalysisRule.

Reply:
@pvillard31 Not sure if you have seen this latest question.  I could make this improvement as I work on NIFI-14132 if you like.

Reply:
Hey @mosermw - sorry for the late reply, I meant to answer your comment and got caught up with other developments. I'm not sure I can get to this during this week so if you're up to it, that would be greatly appreciated!